src/meta/src/barrier/mod.rs

```diff
@@ -174,23 +183,64 @@ pub struct GlobalBarrierManager<S: MetaStore> {
 }
 
 struct CheckpointControl<S: MetaStore> {
-    /// Signal whether the barrier with building actor is sent
-    is_build_actor: bool,
     /// Save the state and message of barrier in order
     command_ctx_queue: VecDeque<EpochNode<S>>,
 }
+/// All changed tables of uncommitted barrier
+#[derive(Default)]
+pub struct ChangedTableId {
+    /// In addition to the actors in `Running`.The barrier need to send or collect the actors of
+    /// these tables.
+    create_table_id: Vec<TableId>,
+    /// The barrier does not send or collect the actors of these tables, even if they are `Running`
+    drop_table_id: Vec<TableId>,
+}
+impl ChangedTableId {
+    /// We will return true and send or collect to the actor, if
+    /// 1. `Running` and not be stop by the previous uncommitted barrier
+    /// 2. no `Running` but create by the previous uncommitted barrier or will create by this
+    /// barrier.
```

Comment:
/// Barrier can be sent to and collected from an actor if:
/// 1. The actor is `Running` and not being dropped.
/// 2. The actor is `Inactive` and belongs to a creating MV.